lib/jmespath/nodes/slice.rb

```diff
@@ -45,7 +46,7 @@ def adjust_slice(length, start, stop, step)
         if step.nil?
           step = 1
         elsif step == 0
-          raise Errors::RuntimeError, 'slice step cannot be 0'
+          return runtime_error('slice step cannot be 0')
```

Comment:
Wouldn't a better improvement be to move this check to the constructor, and make it a syntax-ish error instead?